test/integration/test-amp-skimlinks.js

```diff
@@ -0,0 +1,134 @@
+import {BrowserController, RequestBank} from '../../testing/test-helper';
+import {PLATFORM_NAME} from '../../extensions/amp-skimlinks/0.1/constants';
+import {parseQueryString} from '../../src/url';
+
+const pageTrackingUrl = RequestBank.getUrl('pageTrackingUrl') +
+  '/track.php?data=${data}';
+const linksTrackingUrl = RequestBank.getUrl('linksTrackingUrl') +
+  '/link?data=${data}';
+const nonAffiliateTrackingUrl = RequestBank.getUrl('nonAffiliateTrackingUrl') +
+  '?call=track&data=${data}';
+const waypointUrl = `${RequestBank.getUrl('waypointUrl')}`;
+
+describe('amp-skimlinks', function() {
+  const setup = {
+    extensions: ['amp-skimlinks'],
+    body: `
+            <amp-skimlinks
+                layout="nodisplay"
+                publisher-code="123X123"
+                tracking="true"
+            >
+              <script type="application/json">
+                {
+                    "pageTrackingUrl": "${pageTrackingUrl}",
+                    "linksTrackingUrl": "${linksTrackingUrl}",
+                    "nonAffiliateTrackingUrl": "${nonAffiliateTrackingUrl}",
+                    "waypointUrl": "${waypointUrl}"
+                }
+              </script>
+            </amp-skimlinks>
+            <div>
+                <a id="merchant-link" href="https://nordstrom.com"> Test Merchant </a>
+                <a id="non-merchant-link" href="https://google.com"> Test non-Merchant </a>
+            </div>
+        `,
+  };
+
+  describes.integration('Basic page', setup, env => {
+    let browser = null;
+    let clickLink = null;
+
+    beforeEach(() => {
+      // Simulated click event created by browser.click() does not trigger
+      // the browser navigation when dispatched on a link.
+      // Using MouseEvent("click") instead of Event("click") does,
+      // which is what we want in those tests.
+      clickLink = selector => {
+        const element = env.win.document.querySelector(selector);
+        if (element) {
+          const clickEvent = new MouseEvent('click', {bubbles: true});
+          element.dispatchEvent(clickEvent);
+        }
+      };
+      browser = new BrowserController(env.win);
+      return browser.waitForElementBuild('amp-skimlinks');
+
+    });
+
+    it('Should send the page impression tracking request', () => {
+      return RequestBank.withdraw('pageTrackingUrl').then(req => {
+        const regex = /^\/track\.php\?data=([^&]*)&?.*$/;
+        const match = regex.exec(req.url);
+
+        expect(match).to.have.lengthOf(2);
+        const data = JSON.parse(decodeURIComponent(match[1]));
+        expect(data.jv).to.equal(PLATFORM_NAME);
+        expect(data.pub).to.equal('123X123');
+        // nonblocking.io is the default canonical url
+        expect(data.pag).to.equal('http://nonblocking.io/');
```

Comment:
Would the protocol change depending on the page or is it always `http` ?

Reply:
No it's hardcoded to `http`.